code/go/0chain.net/chaincore/httpclientutil/http_client_util_test.go

```diff
@@ -0,0 +1,1030 @@
+		},
+		{
+			name:    "Empty_URLs_ERR",
+			wantErr: true,
+		},
+		{
+			name: "Response_Status_Not_Ok_ERR",
+			args: args{
+				urls: []string{
+					errServer.URL,
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "Consensus_Success_OK",
+			args: args{
+				urls: []string{
+					server.URL,
+				},
+			},
+			want:    &block.BlockSummary{},
+			wantErr: false,
+		},
+		{
+			name: "Consensus_ERR",
+			args: args{
+				magicBlock: true,
+				urls: []string{
+					server.URL,
+				},
+				consensus: 200,
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := GetBlockSummaryCall(tt.args.urls, tt.args.consensus, tt.args.magicBlock)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GetBlockSummaryCall() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("GetBlockSummaryCall() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestGetMagicBlockCall(t *testing.T) {
+	b := block.NewBlock("", 1)
+	b.HashBlock()
+
+	server := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+				blob, err := json.Marshal(b)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				if _, err := rw.Write(blob); err != nil {
+					t.Fatal(err)
+				}
+			},
+		),
+	)
+	defer server.Close()
+
+	errServer := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+				rw.WriteHeader(http.StatusTooManyRequests)
+			},
+		),
+	)
+	defer errServer.Close()
+
+	errBodyEncroded := httptest.NewServer(
+		http.HandlerFunc(
+			func(rw http.ResponseWriter, r *http.Request) {
+				if _, err := rw.Write([]byte("}{")); err != nil {
+					t.Fatal(err)
+				}
+			},
+		),
+	)
+	defer errBodyEncroded.Close()
+
+	type args struct {
+		urls             []string
+		magicBlockNumber int64
+		consensus        int
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *block.Block
+		wantErr bool
+	}{
+		{
+			name: "Client_ERR",
+			args: args{
+				urls: []string{
+					"wrong url",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name:    "Empty_URLs_ERR",
+			wantErr: true,
+		},
+		{
+			name: "Response_Status_Not_Ok_ERR",
+			args: args{
+				urls: []string{
+					errServer.URL,
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "Decode_Block_ERR",
+			args: args{
+				urls: []string{
+					errBodyEncroded.URL,
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "Consensus_Success_OK",
+			args: args{
+				urls: []string{
+					server.URL,
+				},
+			},
+			want: func() *block.Block {
+				b := block.NewBlock("", 1)
+				b.HashBlock()
+				b.MagicBlock = block.NewMagicBlock()
+
+				return b
+			}(),
+			wantErr: false,
+		},
+		{
+			name: "Consensus_ERR",
+			args: args{
+				urls: []string{
+					server.URL,
+				},
+				consensus: 200,
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Closure error, missing `tt:=tt`.

Reply:
There is no race between goroutines. Test running order is determined by tests struct slice.

Reply:
Fixed.